cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamSpecification.java

```diff
@@ -65,16 +82,16 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof StreamSpecification)) {
```

Comment:
This may create some problematic behavior in future if someone change this class to be non-final.